js/main.js

```diff
@@ -1,128 +1,197 @@
+    }),
+    wifiIcon = new L.Icon({
+      iconUrl: "../images/wifi-color.png",
+      iconSize: [45, 45]
+    }),
+    parksIcon = new L.Icon({
+      iconUrl: "../images/park-color.png",
+      iconSize: [45, 45]
+    }),
+    communityIcon = new L.Icon({
+      iconUrl: "../images/home-color.png",
+      iconSize: [45, 45]
+    });
 
   //create the layer for the map from MapQuest
-  L.tileLayer('http://{s}.tile.osm.org/{z}/{x}/{y}.png', {
-    attribution: '&copy; <a href="http://osm.org/copyright">OpenStreetMap</a> contributors',
-}).addTo( map );
+  L.tileLayer("http://{s}.tile.osm.org/{z}/{x}/{y}.png", {
+    attribution: '&copy; <a href="http://osm.org/copyright">OpenStreetMap</a> contributors'
+  }).addTo(map);
 
-//Convert our JSON files into latitude longitude points in the "Points"
-//variables
-$.get('/datasets/community-centers-cleaned.json',function(data){
-  data.forEach(function(loc){
-    communityPoints.push([+loc.location[1],+loc.location[0]]);
+  function processJSON(data, locationKey) {
+    return data.data.map(function(el) {
+      var lat = el[locationKey][1];
+      var long = el[locationKey][2];
+      var location = ["", ""];
+      if (lat !== null && long !== null) {
+        return [long, lat];
+      }
+      return location;
+    });
+  }
+
+  //Convert our JSON files into latitude longitude points in the "Points"
+  //variables
+  //   fire: getJson.bind(null, 'https://data.nashville.gov/api/views/frq9-a5iv/rows.json'),
+  //   parks: getJson.bind(null, 'https://data.nashville.gov/api/views/74d7-b74t/rows.json'),
+  //   hotspots: getJson.bind(null, 'https://data.nashville.gov/api/views/4ugp-s85t/rows.json'),
+  //   community : getJson.bind(null, 'datasets/community-centers-cleaned.json'),
+  //   police: getJson.bind(null, 'https://data.nashville.gov/api/views/y5ik-ut5s/rows.json')
+  // case "fire":
+  //                               markPoints = processJSON(data, 13, 9);
+  //                               break;
+  //                           case "police":
+  //                               markPoints = processJSON(data, 16, 8);
+  //                               break;
+  //                           case "parks":
+  //                               markPoints = processJSON(data, 41, 8);
+  //                               break;
+  //                           case "hotspots":
+  //                               markPoints = processJSON(data, 11, 8);
+  //                               break;
+  var markPoints;
+  $.get("/datasets/community-centers-cleaned.json", function(data) {
+    data.forEach(function(loc) {
+      communityPoints.push([loc.location[1], loc.location[0]]);
+    });
   });
-});
-$.get('/datasets/firestation-cleaned.json',function(data){
-  data.forEach(function(loc){
-    firePoints.push([+loc.location[1],+loc.location[0]]);
+  $.get("https://data.nashville.gov/api/views/frq9-a5iv/rows.json", function(
+    data
+  ) {
+    markPoints = processJSON(data, 13);
```

Comment:
Instead of using the globally scoped `markPoints` for each of these callbacks, create a new locally scoped variable a different name using `let`, e.g. `let tempPoints = processJSON(data, 13);`